buildkitereporter/buildkitereporter.go

```diff
@@ -0,0 +1,111 @@
+package buildkitereporter
+
+import (
+	"bytes"
+	"context"
+	"fmt"
+	"io"
+	"log"
+	"os"
+	"path/filepath"
+	"strings"
+	"sync"
+
+	"github.com/samsarahq/taskrunner"
+)
+
+func Option(path string) func(*taskrunner.Runtime) {
+	reporter := newReporter(path)
+	return func(r *taskrunner.Runtime) {
+		r.OnStop(func(ctx context.Context, executor *taskrunner.Executor) error {
+			reporter.writeFile(executor.Tasks())
+			return nil
+		})
+		r.Subscribe(func(events <-chan taskrunner.ExecutorEvent) error {
+			for ev := range events {
+				switch event := ev.(type) {
+				case *taskrunner.TaskLogEvent:
+					if event.Stream == taskrunner.TaskLogEventStderr {
+						reporter.write(event.TaskHandler().Definition(), event.Message)
+					}
+				}
+			}
+			return nil
+		})
+	}
+}
+
+type reporter struct {
+	sync.Mutex
+	path    string
+	stderrs map[*taskrunner.Task][]string
+}
+
+var maxLines = 50
+
+// SetMaxLines changes the maximum amount of stderr lines to n.
+func SetMaxLines(n int) { maxLines = n }
```

Comment:
nit, maybe make this an option for the reporter instead of global?

Reply:
Makes sense